### DISCUSSION TITLE
Having the wallet address a part of the idtoken

### DISCUSSION THREAD

**Discussion by shayg-grappa** [Web3Auth/Web3Auth#1306] [Feature Requests]
I'm not sure if that's possible but it would be very useful to have the wallet address be a part of the id token and JWT.

Our use case is a complex onboarding process and it'll be helpful to use the verified JWT for authentication, and base it on wallet address rather than an email.

**Comment by YZhenY** [comment]
Hey there,

We currently provide the public key which can be used to derive an addresses for the chain which you're on

https://betterprogramming.pub/understanding-ethereum-cryptography-3ef7429eddce this might be helpful here

**Comment by ieow** [comment]
you could decode the idToken and get the wallet-> public_key
https://web3auth.io/docs/server-side-verification/social-login-users#sample-idtoken-payload

**Comment by shayg-grappa** [comment]
10x for your help, we did manage to extract the address from the public key

**Comment by grabbou** [comment]
This would be great to document in the server side verification, took a while to find. I guess changing the curve will let obtain other chains' public addresses (such as Solana) in this case.
